doc/nrf/app_approtect.rst

```diff
@@ -0,0 +1,110 @@
+.. _app_approtect:
+
+Enabling access port protection mechanism
+#########################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+.. app_approtect_info_start
+
+Several Nordic Semiconductor SoCs supported in the |NCS| offer an implementation of the access port protection mechanism (APPROTECT).
```

Comment:
Would use AP-Protect as the proper name for the feature, APPROTECT is the name used for related registers.